### PR TITLE
Add option to use domain sockets instead of abstract sockets

### PR DESCRIPTION
**Pull request by dzhulgakov**
Domain sockets are part of file system and, thus, are easier to share in containerized environments. This adds `KINETO_IPC_SOCKET_DIR` env var to control the behavior

### CODE REVIEW

**Review by aaronenyeshi — COMMENTED** [review]
Overall looks good to me, but will leave approval to @briancoutinho , since this code comes from his team.

**Comment by briancoutinho** [comment]
@dzhulgakov thanks for adding this, i'll merge this and also make sure dynolog ipcfabric is synced with this

Also (@aaronenyeshi ) this issue tracks making dynolog a submodule of kineto so we do not have two copies of ipcfabric
https://github.com/facebookincubator/dynolog/issues/57

**Review by aaronenyeshi — APPROVED** [review]
LGTM, approving for landing

File: libkineto/ipcfabric/Endpoint.h [hunk 1]
```diff
@@ -163,8 +174,21 @@ class EndPoint final {
...
   }
 
-  const char* getName(TCtxt const& ctxt) const noexcept {
-    return ctxt.msg_name.sun_path + 1;
+  const char* getName(TCtxt const& ctxt) const {
+    const char* socket_dir = getenv("KINETO_IPC_SOCKET_DIR");
+    bool is_domain_socket = socket_dir && socket_dir[0];
+    if (is_domain_socket) {
+      int prefix_len = strlen(socket_dir) + 1;
+      if (strncmp(socket_dir, ctxt.msg_name.sun_path, prefix_len) == 0) {
```
Comment (briancoutinho) [inline libkineto/ipcfabric/Endpoint.h:182]: @dzhulgakov is this meant to be '!=' 0 instead of == 0

Comment (dzhulgakov) [reply libkineto/ipcfabric/Endpoint.h:182]: It should be, my bad! It didn't trigger because I put `prefix_len` to be +1. C strings are hard :)

**Comment by briancoutinho** [comment]
Thanks for the update @dzhulgakov 
Syncing this up in dynolog in  https://github.com/facebookincubator/dynolog/pull/62

**Pull request closed without merging**